Suppose that, for a certain population, we can predict log earnings from log height as follows:
* A person who is 66 inches tall is predicted to have earnings of $30,000
* Every increase of 1% in height corresponds to a predicted increase of 0.8% in earnings.
* The earnings of approximately 95% of people fall within a factor of 1.1 of predicted values.

a. Give the equation of the regression line and the residual standard deviation of the regression.

b. Suppose the standard deviation of log heights is 5% in this population. What, then, is the $R^2$ of the regression model described here?

### Part A

This is a geometric curve

$$\ln{y} = m\ln{x} + b$$
$$y = e^{m\ln{x}}*e^b$$
$$1.008*y = e^{m\ln{1.01*x}}*e^b$$
$$\frac{1.008*y}{y} = \frac{e^{m\ln{1.01*x}}*e^b}{e^{m\ln{x}}*e^b}$$
$$1.008*e^{m\ln{x}} = e^{m\ln{1.01*x}}$$
$$1.008 = \frac{e^{m\ln{1.01*x}}}{e^{m\ln{x}}}$$
$$1.008 = [\frac{1.01*x}{x}]^m$$
$$1.008 = 1.01^m$$
$$\ln{1.008}=m*\ln{1.01}$$
$$m=\frac{\ln{1.008}}{\ln{1.01}}$$

In [12]:
import numpy as np

m = np.log(1.008)/np.log(1.01)
print(m)

0.800794442593


$$b = \ln{y} - m*\ln{x}$$
$$b = \ln{30000} - 0.800794442593*\ln{66}$$

In [13]:
b = np.log(30000) - m*np.log(66)
print(b)

6.95390042685


$$\ln{y} = 0.801*\ln{x} + 6.95$$

#### Residual Standard Deviation of the Regression

$95%$ of observations fall within $2$ standard errors of the mean. Because $95%$ of observations are said to fall within a factor of $1.1$ of predicted values, then this factor should be equal to $2\times\text{s.e.}$.

A representation of this range is $[\frac{y}{1.1},1.1y]$. The log of the earnings in this range is between $\ln{y}\pm\ln{1.1}$. Because $\ln{y}$ is the mean, then $\ln{1.1} = 2\times\text{s.e.}$.

In [16]:
standard_deviation = np.log(1.1)/2
print(standard_deviation)

0.0476550899022


### Part B
The standard deviation of log heights is 5% in this population. The standard error is $\ln(1.1)/2$. The equation for the $R^2$ is as follows: 
$$R^2 = 1 - \frac{VAR_{res}}{VAR_{tot}}$$

In [21]:
var_residual = np.power(np.log(1.1)/2,2)
var_tot = np.power(.05, 2)

r_squared = 1 - var_residual/var_tot

print(r_squared)

0.0915969625667
